docs: rename "Neon Postgres" to "Lakebase Postgres" (#5408)
neondatabase/website: content/docs/auth/overview.md

### When to use Managed Better Auth vs. self-hosting Better Auth

> Managed Better Auth is a managed authentication service built into the lakebase architecture:

Reviewer: ```suggestion
Managed Better Auth is a managed authentication service built into Lakebase Postgres on Neon:
```
